HeaderLayout.h

```diff
@@ -18,6 +18,7 @@ in the source distribution for its full text.
 
 
 typedef enum HeaderLayout_ {
+   HEADER_LAYOUT_INVALID = -1,
```

Comment:
I'd prefer consistent naming. Apart from this LGTM.

```suggestion
   HF_INVALID = -1,
```